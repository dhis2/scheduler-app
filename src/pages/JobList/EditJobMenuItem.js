import React from 'react'
import { string } from 'prop-types'
import { MenuItem } from '@dhis2/ui-core'
import history from '../../services/history'

const EditJobMenuItem = ({ id }) => (
    <MenuItem dense onClick={() => history.push(`/edit/${id}`)} label="Edit" />
)

EditJobMenuItem.propTypes = {
    id: string.isRequired,
}

export default EditJobMenuItem
