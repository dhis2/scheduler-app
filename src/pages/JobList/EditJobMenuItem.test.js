import React from 'react'
import { shallow, mount } from 'enzyme'
import EditJobMenuItem from './EditJobMenuItem'

beforeEach(() => {
    jest.resetModules()
})

describe('<EditJobMenuItem>', () => {
    it('renders correctly', () => {
        const wrapper = shallow(<EditJobMenuItem id="id" />)

        expect(wrapper).toMatchSnapshot()
    })

    it('calls history.push correctly when MenuItem is clicked', () => {
        const spy = jest.fn()
        jest.doMock('../../services/history', () => ({
            push: spy,
        }))
        const EditJobMenuItem = require('./EditJobMenuItem').default
        const wrapper = mount(<EditJobMenuItem id="id" />)

        wrapper.find('a').simulate('click')

        expect(spy).toHaveBeenCalledWith('/edit/id')
    })
})
