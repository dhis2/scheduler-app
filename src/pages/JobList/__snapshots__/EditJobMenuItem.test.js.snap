// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EditJobMenuItem> renders correctly 1`] = `
<MenuItem
  dataTest="dhis2-uicore-menuitem"
  dense={true}
  label="Edit"
  onClick={[Function]}
/>
`;
